### PR TITLE
ISLANDORA-2218: Add parent column for inactive object management table

### PR DESCRIPTION
**Pull request by BrandonMorr**
**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2218

# What does this Pull Request do?

Add a new configuration option to toggle a 'Parents' column for the Inactive object management list. This new addition should provide helpful links to the parent object(s) for each inactive object.

# What's new?

* Added admin configuration form file admin.inc
* Added islandora_simple_workflow.install to remove configuration variable on module uninstall
* Added menu path 'admin/islandora/tools/simple_workflow/settings', included configuration path in islandora_simple_workflow.info

# How should this be tested?

* Ingest a few different Objects using a variety of cModels (book, large image, pdf, etc.)
* Ensure that each Object's status is set to 'inactive' (may want disable the 'bypass inactive object state' permission to avoid this step)
* In the simple workflow module's settings page enable the parent column.
* Ensure that each inactive object has their parent column information showing up properly (this should include Objects with multiple parents).

Now pick a handful of Objects and set Object viewing XACML restrictions on their parent Object(s) for your test user:
* Refresh the inactive object list and ensure that the parent Objects that you no longer have access to don't show up under the column.

# Interested parties
I can't seem to ping the committers group, so I'll just ping @jordandukart :-)

### CODE REVIEW

**Comment by bondjimbond** [comment]
@BrandonMorr Thanks for this improvement. It's certainly helpful. 

Test results are good. Travis died on the first attempt, so I'll wait till its results come back.

@Islandora/7-x-1-x-committers Any comments on this additional setting?

**Review by rosiel — APPROVED** [review]
Awesome. (and works).

**Review by qadan — APPROVED** [review]
Been through the wringer here internally at DGI, gets my thumbs up 👍

File: includes/admin.inc
Comment (DiegoPino) [inline includes/admin.inc:16]: Wonder if we reeaaaaaally need another toggle. Can this be just shown by default??

File: includes/manage.inc
Comment (DiegoPino) [inline includes/manage.inc:43]: Is there a need for not showing the new parent column? This is a great improvement but a toggle means documentation

**Review by DiegoPino — CHANGES_REQUESTED** [review]
Folks. Do we need another toggle in our lives?

**Comment by DiegoPino** [comment]
Ups. Sorry. I wanted to just comment, not block. But anyway. If I approve now I will forget the conversation I want to start 🧀 .  Why/How showing the new awesome parent column always will make users angry and demand for a toggle? Islandora IS FULL of toggles. Do we need another one for an option that hurts nobody? Toggle == new screenshot == documentation == support.

**Comment by willtp87** [comment]
@DiegoPino This feature will be prohibitively expensive for some situations. As I understand it the form does not page. Paging would be a separate issue that should not be handled here.

**Comment by bondjimbond** [comment]
> This feature will be prohibitively expensive for some situations. As I understand it the form does not page. Paging would be a separate issue that should not be handled here.

At first I agreed with @DiegoPino that a toggle isn't necessary, but @willtp87 makes an interesting point.

How often does such a circumstance come up? I would be surprised to see a repository with so many inactive objects that listing their parents alongside them would eat up an intolerable amount of resources.

**Comment by DiegoPino** [comment]
@BrandonMorr @bondjimbond @rosiel , @willtp87 is right, I understand the tech/performance concern since there is no paging here. I just realized that...
```foreach ($inactive_objects as $inactive_object) ```` hu! , performance will be a concern even withpout the parents if we have a LOT of inactive objects. I remember the issue with making post variables very large in find and replace module because of the same issue, no paging.

So, what do you think? another toggle or add paging?

**Comment by bondjimbond** [comment]
@DiegoPino If the user can configure the number of items in a page, I'd be happy with that -- personally I prefer working with a complete list unless I'm getting a performance hit, so I'd like the option to list all objects.

**Comment by DiegoPino** [comment]
a list of 10000? You are Brave! I will not vote for another toggle/admin option, but I will agree with anything you all agree on. Removing for now the changes request. But I think having a toggle to avoid performance issues is not a good idea, specially if the issues can be avoided completely.

**Comment by bondjimbond** [comment]
> a list of 10000? You are Brave!

@DiegoPino  If it gets that high, I'd enable paging. :)

I think paging is best so long as user can choose how many fit on a page (50, 100, 1000, all?)

**Comment by rosiel** [comment]
This ticket is for a very small improvement, with a nod to potential performance issues, and I think it's fine as it is. I hate toggles too, but i don't think we should block this until 5x the work can be done!  

The "foreach" part is probably fine - even for 10,000 objects - it just populates the table (not saying your browser is gonna be happy, but within that loop there's no object loading unless $render_parents is on). 

Still, I see you, and agree it should be paged, but that deserves a separate ticket.

We could probably get parent info in the sparql query if we're considering performance. And if we go that way, we could include the object owner and the created date and other useful things in the RI...  
so there are a lot of things i'd redesign here if i had my way but this little improvement is good. 

Um yes. It does need to be documented and a new screenshot made though.

**Comment by willtp87** [comment]
@DiegoPino  in https://github.com/Islandora/islandora_simple_workflow/pull/43#issuecomment-387427960 you comment that you are removing the requested changes but you are flagged as requesting changes.  Is this intentional?

**Comment by BrandonMorr** [comment]
@rosiel - Just for clarification, is the documentation blocking this pull request from being merged?

**Review by DiegoPino — APPROVED** [review]

**Comment by DiegoPino** [comment]
@willtp87 confusing from my part. I don't feel an extra toggle here is a good an idea and approving will give the impression I'm ok with merging. So, that. I release the lock, but I'm not an advocate of tiny improvements leaving other issues around.

**Comment by rosiel** [comment]
not a blocker; i'll file a ticket immediately, but PLEASE folks let's start documenting our new features! #documentationAsAFirstClassCitizen

**Comment by DiegoPino** [comment]
@rosiel @bondjimbond  if you are OK with this feel free to move forward. Is the paging/documenting Jira ticket already made? thanks

**Comment by rosiel** [comment]
Documentation ticket: https://jira.duraspace.org/browse/ISLANDORA-2228
Paging ticket: https://jira.duraspace.org/browse/ISLANDORA-2230

**Comment by DiegoPino** [comment]
Thanks @rosiel!!

**Comment by nhart** [comment]
Thank you!

**Comment by BrandonMorr** [comment]
Thanks a lot guys!!

**Pull request merged**
